src/worktree.c

```diff
@@ -405,7 +405,8 @@ int git_worktree_add(git_worktree **out, git_repository *repo,
 		goto out;
 
 	/* Checkout worktree's HEAD */
-	coopts.checkout_strategy = GIT_CHECKOUT_FORCE;
+	if (opts != NULL)
+		memcpy(&coopts, &wtopts.checkout_opts, sizeof(coopts));
```

Comment:
We can guarantee that `wtopts` is validly initialized, even when `opts` is NULL.  I think that we should just:

```suggestion
	memcpy(&coopts, &wtopts.checkout_opts, sizeof(coopts));
```